skyportal/handlers/api/spectrum.py

```diff
@@ -207,6 +218,13 @@ def delete(self, spectrum_id):
         spectrum = Spectrum.query.get(spectrum_id)
         # Permissions check
         _ = Source.get_obj_if_owned_by(spectrum.obj_id, self.current_user)
+
+        # Check that the requesting user owns the spectrum (or is an admin)
+        if not spectrum.is_modifiable_by(self.associated_user_object):
```

Comment:
What happens if the TypeError gets raised?

Reply:
It won't ever get raised as Spectrum will always have `owner`.